frontend/js/actions/getBattleList.js

```diff
@@ -1,11 +1,17 @@
+import { normalize } from 'normalizr';
+
 import { getBattleListPage } from 'utils/api';
 
 import { BATTLE_LIST } from '../constants';
 
+import * as schema from './schema';
+
 function getBattleList() {
   return (dispatch) =>
     getBattleListPage().then((battlesData) => {
-      return dispatch({ type: BATTLE_LIST, payload: battlesData });
+      const normalizedBattleList = normalize(battlesData, schema.battleList);
```

Comment:
I believe you could also do `normalize(battlesData, [schema.battleSchema])`. You probably don't need `schema.battleList`.